File: GameFramework/Assets/Scripts/Framework/GameCore.cs
using System;
using System.IO;
using System.Collections;
using Swift;

/// <summary>
/// 游戏核心
/// </summary>
public class GameCore : Core
{
    private static GameCore instance;
    public static GameCore Instance
    {
        get
        {
            if (instance == null)
                instance = new GameCore();
            return instance;
        }
    }

    // 初始化完成
    public event Action OnInitilized = null;

    // 初始化
    public void Init()
    {
        // 加载配置表
        CsvMgr.Load();

        // 协程组件
        CoroutineManager coMgr = new CoroutineManager();
        Add("CoroutineManager", coMgr);

        // 网络模块
        NetCore nc = new NetCore();
        Add("NetCore", nc);
        // 请求超时时间
        nc.RequestExpireTime = 3;

        // 计时器
        Timer T = new Timer();
        Add("Timer", T);

        // 初始化完成
        if (OnInitilized != null)
            OnInitilized();
    }

}

File: GameFramework/Assets/Scripts/Lua/GameCore.lua
require "event"

GameCore = {}
OnInitilized = event("OnInitilized", true)

function GameCore:New(o)
	o = {}
    setmetatable(o, self)  
    self.__index = self  
	return o
end  

function GameCore:GetInstance()
	if self.m_pGameCore == nil then
		self.m_pGameCore = self:New()
	end
	return self.m_pGameCore
end

function GameCore:ReleaseInstance()
	if self.m_pGameCore then
		self.m_pGameCore = nil
	end
end

-- 初始化
function GameCore:Init()
	print("启动游戏")
	print("正在初始化组件...")
	
	NetCore
	
	OnInitilized:Add(
	function()
		print("加载了某个组件");
	end,self)
	
	-- 初始化完成
	if OnInitilized:Count() ~= 0 then 
		OnInitilized()
	end
	
	print("开始游戏")
end

function Test()

	print("加载了某个组件")

end